### PR TITLE
Bump llvmlite requirement to 0.41.0dev0

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the patch. I think this needs to also update the same places as in #8201 which are noted in the post-release part of the check list in #8841.

**Comment by sklam** [comment]
BFID `numba_yaml_341`

File: buildscripts/incremental/setup_conda_environment.sh
[the remark this answers is not in the document]
Comment (sklam) [reply buildscripts/incremental/setup_conda_environment.sh:1]: ah, i missed that. but should now be fixed in https://github.com/numba/numba/pull/8916/commits/afb5b34edc9a328e7f8692970e4de4306dd54942

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch. This looks about the same as #8201, so I think is as expected.

**Comment by sklam** [comment]
> BFID `numba_yaml_341`

Confirmed that BF used llvmlite-0.41.0dev0. Waiting for the last few tests to finish.

**Comment by sklam** [comment]
> > BFID `numba_yaml_341`
> 
> Confirmed that BF used llvmlite-0.41.0dev0. Waiting for the last few tests to finish.

Passed except for known ARM problems

**Pull request merged**